### PR TITLE
Adds lifecycle events

### PR DESCRIPTION
**Pull request by stevegraham**
Claim lifecycle event logging added for the following events:

- Created
- Login
- Logout
- Deliver access details
- Fee group reference request
- Enqueued
- Received by Jadu
- Rejected by Jadu
- Confirmation email sent
- Payment received
- Payment declined

### CODE REVIEW

**Comment by stevegraham** [comment]
Travis notifications still not coming through :|

**Comment by stevegraham** [comment]
@dan-palmer can i get a review on this when you have a sec please. i'm blocked until i can merge this.

**Comment by dan-palmer** [comment]
@stevegraham oki doki i'll get on to it

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Event < ActiveRecord::Base
+  CREATED                     = 'created'
```
Comment (dan-palmer) [inline app/models/event.rb]: why such not frozen?

File: spec/services/jadu/claim_spec.rb [hunk 2]
```diff
@@ -9,13 +9,17 @@
...
+      hash = { 'feeGroupReference' => '12345678900' }
+      allow(d).to receive(:values_at) { |*k| hash.values_at *k }
+      allow(d).to receive(:[]) { |k| hash[k] }
     end
   end
 
   let(:failure_api_response) do
     double(:failure_api_response, ok?: false).tap do |d|
-      allow(d).to receive(:[])
+      hash = { 'errorCode' => '1', 'errorDescription' => 'herp', 'details' => 'derp' }
```
Comment (dan-palmer) [inline spec/services/jadu/claim_spec.rb]: lol herpes derpes :space_invader:

File: app/models/event.rb (hunk 1 above)
Comment (stevegraham) [inline app/models/event.rb]: much accident

File: spec/models/event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (dan-palmer) [inline spec/models/event_spec.rb]: show him the door? :recycle: :door:

Comment (stevegraham) [inline spec/models/event_spec.rb]: Will add a spec for the readonly functionality.

Comment (dan-palmer) [inline spec/models/event_spec.rb]: :+1:

**Pull request merged**